src/django_components/component.py

```diff
@@ -109,7 +109,8 @@ class RenderInput(Generic[ArgsType, KwargsType, SlotsType]):
 
 
 @dataclass()
-class RenderStackItem(Generic[ArgsType, KwargsType, SlotsType]):
+class MetadataItem(Generic[ArgsType, KwargsType, SlotsType]):
+    render_id: str
```

Comment:
I renamed `RenderStackItem` to `MetadataItem`, hopefully that gives a better idea of what's going on. Also made a tiny change to move the component id from `RenderInput` to `MetadataItem`.

Officially, component's ID is accessed as `Component.id`. But before, one could theoretically access it as `Component.input.id`.

This was not documented, and it also doesn't feel right, because ID is NOT part of the input.